src/components/Referrals/usePagination.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from "react";
+
+type UsePagination = {
```

Comment:
we don't need to define this type, typescript should infer it automatically
But, if you want, i think the name would be PaginationState